coc/static/update_static.py

```diff
@@ -6,229 +6,1482 @@
 import aiohttp
 import asyncio
 import json
-import traceback
-import urllib
-import urllib.request
 import logging
-import zstandard
-import lzma
 import csv
 import os
 import zipfile
-from collections import defaultdict
-
-TARGETS = [
-    ("logic/buildings.csv", "buildings.csv"),
-    ("logic/characters.csv", "characters.csv"),
-    ("logic/heroes.csv", "heroes.csv"),
-    ("logic/pets.csv", "pets.csv"),
-    ("logic/spells.csv", "spells.csv"),
-    ("logic/super_licences.csv", "supers.csv"),
-    ("logic/townhall_levels.csv", "townhall_levels.csv"),
-    ("logic/character_items.csv", "equipment.csv"),
-    ("localization/texts.csv", "texts_EN.csv"),
-]
-
-APK_URL = "https://d.apkpure.net/b/APK/com.supercell.clashofclans?version=latest"
-
-def get_fingerprint():
-    async def download():
-        async with aiohttp.request('GET', APK_URL) as fp:
-            c = await fp.read()
+import zstandard
+import lzma
+from pathlib import Path
+
+class StaticUpdater:
+    def __init__(self):
+
+        self.TARGETS = []
+        self.supported_languages: list[str] = []
+        self.USED_TIDS = set()
+
+        # keep the raw CSV files
+        self.KEEP_CSV = False
+        # keep the raw JSON files
+        self.KEEP_JSON = False
```

Comment:
Those options should be at least accessible via kwargs in the constructor, in my opinion